executor/api/kafka/server.go

```diff
@@ -101,7 +101,22 @@ func NewKafkaServer(fullGraph bool, workers int, deploymentName, namespace, prot
 				producerConfigMap["ssl.certificate.pem"] = sslKakfaServer.ClientCert
 			}
 			producerConfigMap["ssl.key.password"] = sslKakfaServer.ClientKeyPass // Key password, if any
-
+			if util.GetKafkaSecurityProtocol() == "SASL_SSL" {                   //if we also have SASL enabled, then we need to provide the necessary settings in addition to SSL
+				saslKafkaServer := util.GetSaslElements()
+				producerConfigMap["sasl.mechanisms"] = saslKafkaServer.Mechanism
+				if saslKafkaServer.UserName != "" && saslKafkaServer.Password != "" {
+					producerConfigMap["sasl.username"] = saslKafkaServer.UserName
+					producerConfigMap["sasl.password"] = saslKafkaServer.Password
+				}
+			}
+		}
+		if util.GetKafkaSecurityProtocol() == "SASL_PLAIN" { //if we also have SASL enabled, then we need to provide the necessary (no SSL)
```

Comment:
:thought_balloon: To avoid the potential for duplication here, structuring this block of code like the below pseudo-code would help:
```
if protocol == "SSL" || protocol == "SASL_SSL" {
    getSslElements()
    // set up SSL
}
if protocol == "PLAIN" || protocol == "SASL_PLAIN" {
    getSaslElements()
    // set up SASL
}
```

Does that makes sense?  The idea is that instead of nesting logic, we do SSL setup if we need to, and independently SASL setup if we need to.

Reply:
This pattern would then apply to all 3 locations where changes need to happen (currently plaintext is only handled in 2 of them!)

Reply:
Yes, but if the protocol is `PLAIN` without SASL then why grap SASL username and password at all? afaik `PLAIN` mean no authentication mechanism.

Reply:
OK after reading your second comment, I will apply the change (if the docs say so 😉)

Reply:
Done